nan.h

```diff
@@ -1591,39 +1592,51 @@ class Callback {
 
 /* abstract */ class AsyncProgressWorker : public AsyncWorker {
  public:
+
+  struct AsyncData {
+    AsyncData(const char* data_, size_t size_)
+        : data(new char[size_]), size(size_) {
+      memcpy(data, data_, size_);
+    }
+    ~AsyncData() {
+      delete data;
+    }
+    char* data;
+    size_t size;
+  };
```

Comment:
I suppose a case could be made that this is just reinventing `std::string(s, n)`.